MiniAn/minian_parameters.py

```diff
@@ -0,0 +1,335 @@
+
+        self.fr = utils.get_frequency(self.paths[mn_defs.ParametersKeys.FNAME], self.paths[mn_defs.ParametersKeys.H5PATH]+'Time')
+
+        neuron_diameter     = tuple((np.array([self.parameters[mn_defs.ParametersKeys.NEURO_DIAM_MIN], self.parameters[mn_defs.ParametersKeys.NEURO_DIAM_MAX]])/self.parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP]).round().astype('int'))
+        noise_freq: float   = parameters[mn_defs.ParametersKeys.NOISE_FREQ]
+        thres_corr: float   = parameters[mn_defs.ParametersKeys.THRES_CORR]
+
+        advanced_settings = parameters.get(mn_defs.ParametersKeys.ADVANCED_SETTINGS, {})
+
+        # removing advanced_sesttings function keys that are not in the minian functions list
+        minian_functions_list = ["TaskAnnotation", "get_optimal_chk", "custom_arr_optimize",
+                                "save_minian", "open_minian", "denoise", "remove_background",
+                                "seeds_init", "pnr_refine", "ks_refine", "seeds_merge", "initA", "initC",
+                                "compute_trace", "get_noise_fft", "update_spatial", "update_temporal",
+                                "unit_merge", "update_background", "compute_AtC", "apply_transform",
+                                "estimate_motion"] + ["LocalCluster"]
+
+        advanced_settings = {key: advanced_settings[key] for key in advanced_settings if key in minian_functions_list}
+
+        self.params_LocalCluster = {
+            "n_workers": 4,
+            "memory_limit": "auto",
+            "resources": {"MEM": 1}, # constrain the number of tasks that can be concurrently in memory for each worker
+            "threads_per_worker": 2,
+            "dashboard_address": ":8787",
+            "local_directory": self.paths[mn_defs.ParametersKeys.TMP_DIR]
+        }
+        if "LocalCluster" in advanced_settings:
+            advanced_settings["LocalCluster"] = {key: advanced_settings["LocalCluster"][key] for key in advanced_settings["LocalCluster"] if key in self.params_LocalCluster}
+            self.params_LocalCluster.update(advanced_settings["LocalCluster"])
+
+        self.params_load_doric = {
+            "fname": self.paths[mn_defs.ParametersKeys.FNAME],
+            "h5path": self.paths[mn_defs.ParametersKeys.H5PATH],
+            "dtype": np.uint8,
+            "downsample": {"frame": parameters["TemporalDownsample"],
+                            "height": parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP],
+                            "width": parameters[mn_defs.ParametersKeys.SPATIAL_DOWN_SAMP]},
+            "downsample_strategy": "subset",
+        }
+
+        self.params_save_minian = {
+            "dpath": os.path.join(self.paths[mn_defs.ParametersKeys.TMP_DIR], "final"),
+            "meta_dict": {"session": -1, "animal": -2},
+            "overwrite": True,
+        }
+
+        self.params_get_optimal_chk = {
+            "dtype": float
+        }
+        if "get_optimal_chk" in advanced_settings:
```

Comment:
I would predefine all the parameters and then call the `update_func_params()` in a loop:
```
for func_name in advanced_wettings:
    self.update_func_params(...)
```
And add a check if function name is valid
Does it make sense?

Reply:
I can see what I can do.
But that is a bit tricky... because I need the function (not the name but the function type to check all the arguments but I can check what I can do)